### ISSUE TITLE
(Feature request) Folding

### ISSUE THREAD

**Issue by s0h3ck** [zadam/trilium#2470]
### Describe feature

Hello, I would like to have the possibility to folder section of a document ;

To get started with this idea, let's start looking at two type of folding I would like to see implemented: 

Type 1 : Header Folding
Explanation: The user creates an h1 h2 h3 h1 h2 h3 h1 item. When you folder the header h1. You only see h1-h1 h2 h3 h1. Hope that make sense. See vim folding to understand what I mean.

Type 2: Code block Folding
Explanation: Similar to Type 1, but only for the code block

Note: These two types of folding should be remembered when you close or switch from one document to another.

**Comment by zadam** [comment]
Hi, this would have to be implemented in CKEditor, but [the corresponding request](https://github.com/ckeditor/ckeditor5/issues/1504) doesn't give much hope. This doesn't mean it's impossible, but the required effort would be higher.

**Comment by s0h3ck** [comment]
Ha I see, curiosity, which part is this folding attached to (see image down below)?

![2021-12-27_21-55](https://user-images.githubusercontent.com/7539558/147523011-2649bfc7-4bf9-400f-94d2-4eff06c5a53e.png)

There is definitively an arrow that hide or unhide a note. Can that feature be moved into a note?

**Comment by zadam** [comment]
No, this is in search. Having this in the CKEditor is a completely different thing, probably also much more difficult :-/

**Comment by zadam** [comment]
Related #2470

**Comment by meichthys** [comment]
Duplicate of #3636
